packages/gatsby/index.d.ts

```diff
@@ -1539,3 +1538,17 @@ export interface Page<TContext = Record<string, unknown>> {
   component: string
   context: TContext
 }
+
+export interface IPluginRefObject {
+  resolve: string
+  options?: IPluginRefOptions
+  parentDir?: string
+}
+
+export type PluginRef = string | IPluginRefObject
```

Comment:
Is this type different from the one defined in `src/bootstrap/load-plugins/types`?

If they are not the same, could they please have a different, unique, name? If they are, could you replace the definition in that file instead?